bundles/binding/org.openhab.binding.onkyo/src/main/java/org/openhab/binding/onkyo/internal/eiscp/EiscpSerial.java

```diff
@@ -0,0 +1,258 @@
+        notifier = new EispEventNotifier(serialPort);
+    }
+
+    @Override
+    public synchronized void addEventListener(OnkyoEventListener listener) {
+        notifier.addEventListener(listener);
+    }
+
+    @Override
+    public synchronized void removeEventListener(OnkyoEventListener listener) {
+        notifier.removeEventListener(listener);
+    }
+
+    @Override
+    public int getRetryCount() {
+        return retryCount;
+    }
+
+    @Override
+    public void setRetryCount(int retryCount) {
+        this.retryCount = retryCount;
+    }
+
+    @Override
+    public boolean connectSocket() {
+        return connect(serialPortName);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public boolean connect(String serialPortName) {
+
+        try {
+            logger.debug("Open connection to serial port '{}'", serialPortName);
+            CommPortIdentifier portIdentifier = CommPortIdentifier.getPortIdentifier(serialPortName);
+
+            CommPort commPort = portIdentifier.open(this.getClass().getName(), 2000);
+
+            serialPort = (SerialPort) commPort;
+            serialPort.setSerialPortParams(9600, SerialPort.DATABITS_8, SerialPort.STOPBITS_1, SerialPort.PARITY_NONE);
+            serialPort.enableReceiveThreshold(1);
+            serialPort.disableReceiveTimeout();
+
+            in = serialPort.getInputStream();
+            out = serialPort.getOutputStream();
+
+            out.flush();
+            if (in.markSupported()) {
+                in.reset();
+            }
+
+            // RXTX serial port library causes high CPU load
+            // Start event listener, which will just sleep and slow down event loop
+            serialPort.addEventListener(this);
+            serialPort.notifyOnDataAvailable(true);
+            return true;
+        } catch (Exception e) {
+            logger.error("serial port connect error", e);
+            e.printStackTrace();
+            return false;
+        }
+
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean closeSocket() {
+        try {
+            if (out != null) {
+                logger.debug("Close serial out stream");
+                IOUtils.closeQuietly(out);
+            }
+            if (in != null) {
+                logger.debug("Close serial in stream");
+                IOUtils.closeQuietly(in);
+            }
+            if (serialPort != null) {
+                logger.debug("Close serial port");
+                serialPort.close();
+            }
+
+            serialPort.removeEventListener();
+
+            serialPort = null;
+            out = null;
+            in = null;
+
+            logger.debug("Closed");
+            return true;
+        } catch (Exception e) {
+            logger.error("Unable to close ", e);
+            return false;
+        }
+    }
+
+    private StringBuilder getEiscpMessage(String eiscpCmd) {
+        logger.debug("Requested Command is " + eiscpCmd);
```

Comment:
Definitely use parameterized logging here instead of string concatenation, because otherwise this method will be generating garbage objects on every call!